### PR TITLE
[CVE Fix] Fixes CVE-2022-42920 by forcing bcel version to resovle to 6.6

### LINKED ISSUE #2248: CVE-2022-42920 (High) detected in bcel-6.5.0.jar
## CVE-2022-42920 - High Severity Vulnerability
<details><summary> Vulnerable Library - <b>bcel-6.5.0.jar</b></p></summary>

<p>Apache Commons Bytecode Engineering Library</p>
<p>Library home page: <a href="https://commons.apache.org/proper/commons-bcel">https://commons.apache.org/proper/commons-bcel</a></p>
<p>Path to dependency file: /build.gradle</p>
<p>Path to vulnerable library: /home/wss-scanner/.gradle/caches/modules-2/files-2.1/org.apache.bcel/bcel/6.5.0/79b1975ec0c7a6c1a15e19fb3a58cc4041b4aaea/bcel-6.5.0.jar</p>
<p>

Dependency Hierarchy:
  - spotbugs-4.7.3.jar (Root Library)
    - :x: **bcel-6.5.0.jar** (Vulnerable Library)
<p>Found in base branch: <b>main</b></p>
</p>
</details>
<p></p>
<details><summary> Vulnerability Details</summary>
<p>  
  
Apache Commons BCEL has a number of APIs that would normally only allow changing specific class characteristics. However, due to an out-of-bounds writing issue, these APIs can be used to produce arbitrary bytecode. This could be abused in applications that pass attacker-controllable data to those APIs, giving the attacker more control over the resulting bytecode than otherwise expected. Update to Apache Commons BCEL 6.6.0.

<p>Publish Date: 2022-11-07
<p>URL: <a href=https://www.mend.io/vulnerability-database/CVE-2022-42920>CVE-2022-42920</a></p>
</p>
</details>
<p></p>
<details><summary> CVSS 3 Score Details (<b>9.8</b>)</summary>
<p>

Base Score Metrics:
- Exploitability Metrics:
  - Attack Vector: Network
  - Attack Complexity: Low
  - Privileges Required: None
  - User Interaction: None
  - Scope: Unchanged
- Impact Metrics:
  - Confidentiality Impact: High
  - Integrity Impact: High
  - Availability Impact: High
</p>
For more information on CVSS3 Scores, click <a href="https://www.first.org/cvss/calculator/3.0">here</a>.
</p>
</details>
<p></p>
<details><summary> Suggested Fix</summary>
<p>

<p>Type: Upgrade version</p>
<p>Origin: <a href="https://lists.apache.org/thread/lfxk7q8qmnh5bt9jm6nmjlv5hsxjhrz4">https://lists.apache.org/thread/lfxk7q8qmnh5bt9jm6nmjlv5hsxjhrz4</a></p>
<p>Release Date: 2022-11-07</p>
<p>Fix Resolution: org.apache.bcel:bcel:6.6.0</p>

</p>
</details>
<p></p>

### PR DESCRIPTION
**Pull request by DarshitChanpura**
* Category: Bug fix

### Issues Resolved
Resolves #2248 

### Check List
~- [ ] New functionality includes testing~
~- [ ] New functionality has been documented~
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by cwperks** [comment]
Is there any way to track when a new spotbugs gradle plugin is release? https://plugins.gradle.org/plugin/com.github.spotbugs

In this discussion with the maintainer of spotbugs I see that 4.7.4 has not been released yet and 4.7.4 contains an update for bcel from 6.5.0 -> 6.6.1: https://github.com/spotbugs/spotbugs/discussions/2251

Once 4.7.4 is released the gradle plugin also needs to follow-up with a release.

**Comment by DarshitChanpura** [comment]
> Is there any way to track when a new spotbugs gradle plugin is release? https://plugins.gradle.org/plugin/com.github.spotbugs
> 
> In this discussion with the maintainer of spotbugs I see that 4.7.4 has not been released yet and 4.7.4 contains an update for bcel from 6.5.0 -> 6.6.1: [spotbugs/spotbugs#2251](https://github.com/spotbugs/spotbugs/discussions/2251)
> 
> Once 4.7.4 is released the gradle plugin also needs to follow-up with a release.

I'm not sure about the way to track the release automatically but I will keep an eye out for 4.7.4, and will update our build.gradle once it is out.

File: build.gradle [hunk 1]
```diff
@@ -262,6 +262,7 @@ configurations {
             force "io.netty:netty-handler:${versions.netty}"
             force "io.netty:netty-transport:${versions.netty}"
             force "io.netty:netty-transport-native-unix-common:${versions.netty}"
+            force "org.apache.bcel:bcel:6.6.0" // Fixes `issues/2248`. This dependency comes transitively via spotbugs
```
Comment (peternied) [inline build.gradle:265]: Can we remove this comment, having this information in the commit message should be enough

Comment (DarshitChanpura) [reply build.gradle:265]: I kept this here as a reminder to remove it once spotbugs releases a patch with this fix. Let me update the comment to say as such

**Review by peternied — APPROVED** [review]

**Review by cwperks — APPROVED** [review]

**Pull request merged**